oci/repositories.bzl

```diff
@@ -37,8 +39,81 @@ crane_repositories = repository_rule(
     },
 )
 
+
+ZOT_BUILD_TMPL = """\
+# Generated by container/repositories.bzl
+load("@contrib_rules_oci//oci:toolchain.bzl", "registry_toolchain")
+registry_toolchain(
+    name = "zot_toolchain", 
+    registry = "zot",
+    launcher = "launcher.sh"
+)
+"""
+
+ZOT_LAUNCHER_TMPL = """\
+#!/usr/bin/env bash
+set -o errexit -o nounset -o pipefail
+
+SCRIPT_DIR="$(dirname "${BASH_SOURCE[0]}")"
+ZOT="${SCRIPT_DIR}/zot"
+
+function start_registry() {
+    local CONFIG_PATH="$1/config.json"
+    cat > "${CONFIG_PATH}" <<EOF
+{
+    "storage": {"rootDirectory": "$1" },
+    "http": { "port": "0", "address": "127.0.0.1" },
+    "log": { "level": "info" }
+}
+EOF
+
+    local OUTPUT="$2"
+
+    "${ZOT}" serve "${CONFIG_PATH}" 2>&1 >> "${OUTPUT}" &
+
+    local DEADLINE=5
```

Comment:
this feels like a workaround, should we file a bug on something like zot so we don't have to be the ones implementing a timeout?

Reply:
actually, this is not zot's fault. even a simple HTTP server will take some `ms` to come alive. there is nothing zot can do here to help us.

Reply:
we should even do more here by checking if zot actually started accepting the requests on the port assigned to it by the kernel.